src/patterns/ec2-app.ts

```diff
@@ -94,51 +125,108 @@ function restrictedCidrRanges(ranges: IPeer[]) {
   }));
 }
 
-/*
- * This pattern is under development. Please don't attempt to use it yet.
+/**
+ * Pattern which creates all of the resources needed to run an application on EC2 instances.
+ * For convenience, you may wish to use [[`GuPlayApp`]] or [[`GuNodeApp`]], which extend this class.
  *
- * Usage:
+ * This pattern will grant your EC2 instances a number of commonly needed IAM permissions. For more information on
+ * this, see [[`GuInstanceRole`]]. To add additional permissions to your EC2 instances, see [[`GuEc2AppProps`]].
+ *
+ * This pattern will automatically create security groups/rules which allow for:
+ * 1. Incoming traffic over HTTPS (from the whole internet or from the specified CIDR ranges, depending on the
+ * [[`AppAccess`]] specified).
+ * 2. Communication between the load balancer and the EC2 instances over HTTP, via the specified application port.
+ * 3. Outbound traffic from your EC2 instances over HTTPs (to enable communication with third-party APIs).
+ *
+ * The pattern will run a single EC2 instance in `CODE` and three EC2 instances in `PROD`.
+ *
+ * Example usage for a public facing app (open to the internet):
  * ```typescript
  * new GuEc2App(stack, {
- *       applicationPort: GuApplicationPorts.Node,
- *       app: "app-name",
- *       // ONLY allows access to CIDR ranges provided
- *       access: { scope: AccessScope.RESTRICTED, cidrRanges: [Peer.ipv4("192.168.1.1/32"), Peer.ipv4("8.8.8.8/32")] },
- *       certificateProps:{
- *           [Stage.CODE]: {
- *             domainName: "code-guardian.com",
- *             hostedZoneId: "id123",
- *           },
- *           [Stage.PROD]: {
- *             domainName: "prod-guardian.com",
- *             hostedZoneId: "id124",
- *           },
- *       },
- *       monitoringConfiguration: { noMonitoring: true },
- *       userData: "",
+ *   applicationPort: 1234,
+ *   app: "app-name",
+ *   access: { scope: AccessScope.PUBLIC },
+ *   certificateProps:{
+ *     [Stage.CODE]: {
+ *       domainName: "code-guardian.com",
+ *       hostedZoneId: "id123",
+ *     },
+ *     [Stage.PROD]: {
+ *       domainName: "prod-guardian.com",
+ *       hostedZoneId: "id124",
+ *     },
+ *   },
+ *   monitoringConfiguration: {
+ *     tolerated5xxPercentage: 1,
+ *     snsTopicName: "alerts-topic-for-my-team",
+ *   },
+ *   userData: {
+ *     distributable: {
+ *       fileName: "app-name.deb",
+ *       executionStatement: `dpkg -i /app-name/app-name.deb`,
+ *     }
+ *   },
  * });
+ * ```
  *
- * // For public-facing applications
+ * Example usage for an app which is locked down to specific IP ranges:
```

Comment:
minor: should we use the same terminology as the interface? "locked down" -> "restricted", "IP ranges" -> "CIDR ranges"?